src/components/ExpensiTextInput/BaseExpensiTextInput.js

```diff
@@ -145,22 +160,43 @@ class BaseExpensiTextInput extends Component {
                                 labelScale={this.state.labelScale}
                             />
                         ) : null}
-                        <TextInput
-                            ref={(ref) => {
-                                if (typeof innerRef === 'function') { innerRef(ref); }
-                                this.input = ref;
-                            }}
+                        <View style={{display: 'flex', flexDirection: 'row'}}>
+                            <TextInput
+                                ref={(ref) => {
+                                    if (typeof innerRef === 'function') { innerRef(ref); }
+                                    this.input = ref;
+                                }}
                             // eslint-disable-next-line
                             {...inputProps}
-                            value={value}
-                            placeholder={(this.state.isFocused || !label) ? placeholder : null}
-                            placeholderTextColor={themeColors.placeholderText}
-                            underlineColorAndroid="transparent"
-                            style={inputStyle}
-                            onFocus={this.onFocus}
-                            onBlur={this.onBlur}
-                            onChangeText={this.setValue}
-                        />
+                                value={value}
+                                placeholder={(this.state.isFocused || !label) ? placeholder : null}
+                                placeholderTextColor={themeColors.placeholderText}
+                                underlineColorAndroid="transparent"
+                                style={[...inputStyle, {flex: 1, paddingTop: 13}]}
+                                onFocus={this.onFocus}
+                                onBlur={this.onBlur}
+                                onChangeText={this.setValue}
+                                secureTextEntry={this.state.passwordHidden}
+                            />
+                            {secureTextEntry && (
+                            <Pressable
+                                accessibilityRole="button"
+                            // eslint-disable-next-line lines-around-comment
+                            // accessibilityLabel={this.props.translate('sidebarScreen.buttonSearch')}
```

Comment:
To make this label work I'd have to wrap this in withLocalization which would mean all inputs would instantiate the HoC even if they don't need any translations.
Would this be an issue?